core/drivers/stm32_i2c.c

```diff
@@ -753,6 +754,8 @@ int stm32_i2c_init(struct i2c_handle_s *hi2c,
 	vaddr_t base = 0;
 	uint32_t val = 0;
 
+	mutex_pm_aware_init(&hi2c->mu);
```

Comment:
Actually, shouldn't this be done at probe instead of this exported function?

Reply:
We need to initialize the mutex from `stm32_i2c_init()` because this is the init function used by an Stm32 I2C device that do not rely on DT_DRIVER driver probing support, see _[core/drivers/crypto/se050/glue/i2c_stm32.c](https://github.com/OP-TEE/optee_os/blob/4.1.0/core/drivers/crypto/se050/glue/i2c_stm32.c#L89)_. Note that `stm32_i2c_init()` is called from the probe function, as you expect.

Reply:
Then stm32_i2c_init() should be called once? Can we protect against several use of this function per i2c handler?